### ISSUE TITLE
notebook preview is only rendering for *owner*

### ISSUE THREAD

**Issue by RZachLamberty** [microsoft/AzureNotebooks#178]
### Azure Notebooks environment 

I accessed the notebook environment via notebooks.azure.com. Notebooks are developed in a jupyter notebook server (jupyter v 1.0.0, notebook v 5.0.0) on a debian (Jessie) linux server, and pulled in from git (via a terminal).

### Description

I can load and execute notebooks -- everything works fine. Other users can open the links to these notebooks, but nothing is rendered.

### Repro Steps

I am unaware of exactly how isolated the problem is to me; the following is reproducible (but obviously only for my notebooks)

1. be anyone other than me (I can do this if I simply log out as well)
2. click on any of the ipynb files in library `gu511`

I have no way of verifying if this happens for notebooks created by other users.

### Usage Scenario

I am teaching a course at Georgetown and using this service as a lecture notes host.

### Recommended Metadata
The Jupyter Kernel (Python2, Python3, R, N/A): Python 3
Occurrence Time: since 2017.09.13 at least
Web Browser: verified in chrome and IE
LibraryID and Notebook Name (if applicable): library gu511, any notebook (specifically: 004_python.ipynb)
Logs (!cat .nb.log Or: New/Terminal, $cat .nb.log) - attach files if applicable

**Comment by smortaz** [comment]
Thanks for filing this bug!  it does repro for me as well.  

Team: all the notebooks are live/reveal slides.  none render.  

we'll take a look.  the only workaround right now is make them non-slide mode notebooks, or for ppl to clone & run the slide notebooks.   we'll update here as soon as it's fixed.

**Comment by denfromufa** [comment]
@RZachLamberty can't your students clone your notebooks?

**Comment by smortaz** [comment]
they can i'm sure, but it should not be necessary - slide mode should render fine.  it's a legit bug. 

also, slide mode w no login is *very* useful.   pushing ppl to login just to view slides is too much... login is required if you plan to edit/run/etc.

**Comment by RZachLamberty** [comment]
just to be clear, I was the only person launching them in the reveal slide mode previously -- when I say they rendered, I mean the static html version that is simply the rendered cells in normal notebook format (e.g. how they would appear on github).

having the students able to launch the reveal slides is not a requirement for me, but being able to launch the slides as a reveal presentation myself *is* a requirement (it's how I present them in class), so stripping the cell metadata info is going to be a problem.

**Comment by smortaz** [comment]
understood.  you can view them in normal mode if you're logged in, or click
them little "chart" icon and go into reveal mode and do a full slide
presentation.  i tested that and it works (btw, great slides!)

the only thing that is broken is html preview w/o being logged in.
 something is different about /these/ slides as other reveal slide
notebooks render fine w/o being logged in.  we'll look it after the wknd.

**Comment by RZachLamberty** [comment]
One thought: this may have occurred after a recent upload of notebooks that
imported plotly and initializes the notebooks in offline mode. Only one
notebook is doing this and I don't think I have yet executed those cells on
an azure server, but I know that the initialization attempts to download
and inject plotly js libraries and perhaps that has tripped some failsafe?

**Comment by smortaz** [comment]
that's entirely possible. can you comment out that/those cells and see if
it renders properly?  i know plot.ly folks are working on a rendering issue
which is security related.

**Comment by RZachLamberty** [comment]
I did the following to test the behavior of plotly:

- created a new library and notebook in which identical plotly actions render consistently
- commented out all plotly cells from gu511 library 004_python.ipynb (does not render)
- completely removed all plotly cells from gu511 library 004_python.ipynb (does not render)

in my next comment I'll discuss what I think is *actually* going on, but I'm leaning toward plotly being (at worst) a contributor to the problem

**Comment by RZachLamberty** [comment]
I noticed that one action will take an un-renderable notebook and make it renderable: delete html `<img>` elements from markup cells.

for example, in library gu511, notebook 006_cli_boto.ipynb will be un-renderable as is, but the act of deleting the one `<img>` tag in that notebook will make it renderable.

I decided to create a test library (gu511-test) with the following items:

1. `works.ipynb` -- the `plotly` items render correctly
2. `works_nb.ipynb` -- an intra-domain `<img>` tag
3. `works_imgr.ipynb` -- an inter-domain `<img>` tag
4. `aint_works_10_img.ipynb` -- just ten identical `<img>` tag elements in consecutive cells

the first three will render as expected, and the 4th one will fail. 

Given that *some* `<img>`-containing docs render, I am wondering if the act of generating the `html` extracts has a certain timeout set, and that slow responses from the `src` are causing that timeout to be met.
